tests/tests_lite/plugins/collectives/test_torch_collective.py

```diff
@@ -272,6 +304,16 @@ def _test_two_groups(strategy, left_collective, right_collective):
 
 
 @skip_distributed_unavailable
-@pytest.mark.skip(reason="TODO(carmocca): causing hangs in CI")
-def test_two_groups():
-    collective_launch(_test_two_groups, [torch.device("cpu")] * 3, num_groups=2)
+@pytest.mark.parametrize(
+    ["device_type", "autosetup_strategy"],
+    [
+        ("cpu", True),
+        ("cpu", False),
+        pytest.param("cuda", True, marks=RunIf(min_cuda_gpus=3)),
+        pytest.param("cuda", False, marks=RunIf(min_cuda_gpus=3)),
+    ],
+)
+@mock.patch.dict(os.environ, os.environ.copy(), clear=True)  # sets CUDA_MODULE_LOADING in torch==1.13
+@RunIf(skip_windows=True)  # flaky test on Windows - RuntimeError: Socket Timeout
```

Comment:
I experimented a bit more but could not find the cause of it. I don't think we should invest more time in this at the moment.

Reply:
Thank you, @awaelchli for the help with debugging this anyway! I fully agree with you, too much time was spent on this already.